### ISSUE TITLE
Not resolving on OpenWrt 18.06.1

### ISSUE THREAD

**Issue by iwattud** [DNSCrypt/dnscrypt-proxy#575]
Hello,

I have a TP-Link Archer C7 v2 which was running an older version of dnscrypt-proxy.
My config was running LEDE 17.01 "Reboot" and my dnscrypt-proxy config was more or less the one in https://wiki.openwrt.org/inbox/dnscrypt, (but with three instances)

Yesterday I updated my firmware to OpenWrt 18.06.1 and I followed step y step this installation guide:
https://github.com/jedisct1/dnscrypt-proxy/wiki/Installation-on-OpenWRT, but then no one of the devices on my LAN was able to resolve any name.

I remember that with the old dnscrypt-proxy release (maybe dnscrypt-proxy 1.9.5, not sure about that) I needed to modify some files on the /etc/config/, such as dhcp and dnscrypt-proxy, and it seems those entries are not needed to run dnscrypt-proxy 2.

Can be the entries modified on the /etc/config/dhcp and /etc/config/dnscrypt-proxy interfering with the proper operation of dnscrypt-proxy 2?

**Comment by jedisct1** [comment]
Follow the steps from the documentation again...

I have exactly the same router, running OpenWRT, and it works when exactly following these steps, on a vanilla installation.

Maybe you have some leftovers from a previous installation? Try to undo any changes made from the old install.

**Comment by iwattud** [comment]
Thank you for your fast response.
I did all the steps again but i got the same results.
I used the file dnscrypt-proxy-linux_mips-2.0.16.tar.gz. I suppose it is the correct one for my (our) router.

Taking a look for "legacy" config files I found this:

I needed to remove those entries from /etc/config/dhcp that were working with the old dnscrypt-proxy
list server '127.0.0.1#2222'
list server '127.0.0.1#3333'
list server '127.0.0.1#4444'
and I added this (not sure if necessary)
list server '127.0.0.53#53'

Also in the /etc/config/network. I needed to change on config interface 'lan' the option dns '127.0.0.1' for option dns '127.0.0.53'

It seems now the devices on my network are resolving using the router IP address as resolver.

Thanks!
